.github/workflows/auto_branching.yml

```diff
@@ -0,0 +1,228 @@
+### The auto-branching workflow triggered through a dispatch request from the CI
+name: auto-branching
+
+# Run on workflow dispatch from CI
+on:
+  workflow_dispatch:
+    inputs:
+      target_branch:
+        type: string
+        description: branch to be created from the master
+      stream_version:
+        type: string
+        description: new stream version of satellite
+
+jobs:
+  auto-branching-non-master:
+    name: ${{ github.event.inputs.target_branch }} - raise PR with changes
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Create the ${{ github.event.inputs.target_branch }} branch
+        id: create-branch
+        uses: peterjgrainger/action-create-branch@v2.2.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
BTW, We need to create a new token, because probably CHERRYPICK_PAT token does not have access for all that we do here. Especially, 
- creating new labels
- adding branch protection rule

@JacobCallahan Could you check that CHERRYPICK_PAT has these 2 permissions? If not we have to create a new token for these admin kind of roles!